pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -109,10 +109,17 @@ private static String removeTerminatingSemicolon(String sql) {
 
   public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql)
       throws SqlCompilationException {
+    return compileToSqlNodeAndOptions(sql, true);
+  }
+
+  public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql, boolean isRemoveComments)
+      throws SqlCompilationException {
     long parseStartTimeNs = System.nanoTime();
 
     // Remove the comments from the query
```

Comment:
The code for removing comments was added in https://github.com/apache/pinot/issues/7714 to avoid misusing the older support for `OPTIONS`. As per my understanding, this logic is not needed anymore because we have moved to SET based options. 

So we can remove this logic entirely?